src/stat.go

```diff
@@ -44,7 +51,24 @@ func (g *Commands) statfn(fname string, fn func(string) (*File, error)) error {
 			continue
 		}
 
-		err = g.stat(src, f, g.opts.Depth)
+		if g.opts.Md5sum {
+
+			depth := g.opts.Depth
+			src = f.Name // forces filename if -id is used
+
+			// md5sum with no arguments should do md5sum *
+			if f.IsDir && g.opts.Path == "/" {
+				src = ""
+				if depth == 1 {
```

Comment:
Hmm, this only catches the case when you run it from root. To ensure that a depth at least of 2 when 1 is set is used:
```diff
diff --git a/src/stat.go b/src/stat.go
index b86e21f..a032c1e 100644
--- a/src/stat.go
+++ b/src/stat.go
@@ -57,8 +57,10 @@ func (g *Commands) statfn(fname string, fn func(string) (*File, error)) error {
                        src = f.Name // forces filename if -id is used
 
                        // md5sum with no arguments should do md5sum *
-                       if f.IsDir && g.opts.Path == "/" {
-                               src = ""
+                       if f.IsDir {
+                               if g.opts.Path == "/" {
+                                       src = ""
+                               }
                                if depth == 1 {
                                        depth = 2
                                }
}
```

Even better, I've fixed the depth limiting restraint in commit https://github.com/odeke-em/drive/commit/a9db767311995686b188853eceddd5a6acdd1a20 and if you get that patch in, all you'll have to do then is:

```diff
diff --git a/src/stat.go b/src/stat.go
index b86e21f..634b0ab 100644
--- a/src/stat.go
+++ b/src/stat.go
@@ -57,11 +57,8 @@ func (g *Commands) statfn(fname string, fn func(string) (*File, error)) error {
                        src = f.Name // forces filename if -id is used
 
                        // md5sum with no arguments should do md5sum *
-                       if f.IsDir && g.opts.Path == "/" {
+                       if f.IsDir && rootLike(g.opts.Path) {
                                src = ""
-                               if depth == 1 {
-                                       depth = 2
-                               }
                        }
```